src/eureka/S4_generate_lightcurves/s4_genLC.py

```diff
@@ -200,7 +200,32 @@ def genlc(eventlabel, ecf_path=None, s3_meta=None, input_meta=None):
                 meta.nplots = meta.n_int
 
             # Determine wavelength bins
-            if meta.nspecchan is None:
+            if meta.wave_input is not None:
+                # bins defined by file input. 2 columns: low and high edges
+                meta.wave_low, meta.wave_hi = np.genfromtxt(meta.wave_input).T
+                meta.wave = (meta.wave_low + meta.wave_hi)/2
+                meta.nspecchan = len(meta.wave)
+                log.writelog(f'  Using input file to create {meta.nspecchan} '
+                             'channels.')
+            elif meta.nspecchan is None and meta.npixelbins is not None:
+                # User wants bins defined by the given number of pixels
+                istart = np.where(wave_1d >= meta.wave_min)[0][0]
+                iend = np.where(wave_1d >= meta.wave_max)[0][0]
+                # Shift bins by some number of pixels (only useful for MIRI)
+                istart += meta.npixelshift
+                iend += meta.npixelshift
+                edges = wave_1d[istart:iend+meta.npixelbins:meta.npixelbins]
+                dwav = np.ediff1d(
+                    wave_1d[istart:iend+meta.npixelbins])[::meta.npixelbins]/2
```

Comment:
Please double-check that this is correct. At first glance, this doesn't look like it would work and it seems to me like you would want to move the `[::meta.npixelbins]` inside of the parentheses; otherwise, the ediff1d would be the difference in wavelength from one pixel to the next, rather than from one bin to the next